### PR TITLE
API-13744: Insurance Buffer IT configured to run locally

### PR DESCRIPTION
**Pull request by jabando-va**

### CODE REVIEW

File: vista-fhir-query-mock-services/src/main/java/gov/va/api/health/vistafhirquery/mockservices/LhsLighthouseRpcGatewayMocksV1.java [hunk 1]
```diff
@@ -34,38 +38,85 @@
...
+        .findFirst()
+        .orElseThrow(() -> new IllegalStateException("No response for: " + ap));
+  }
+
+  @SuppressWarnings("UnnecessaryParentheses")
+  private static String writeResponse(RpcDetails.Parameter ap) {
+    var results =
+        ap.array().stream()
+            .filter("api^create^coverage"::equals)
+            .map(p -> p.replace("api^create^", ""))
```
Comment (jhulbert-va) [inline vista-fhir-query-mock-services/src/main/java/gov/va/api/health/vistafhirquery/mockservices/LhsLighthouseRpcGatewayMocksV1.java:98]: This filter will always return `coverage` and nothing else. I think you'll want a pattern matcher for this `Pattern.compile("([0-9\\.]+\\^([0-9]+)\\^#.*)")` and then check for the fileman strings. You've already verified it's a write.

File: vista-fhir-query-mock-services/src/main/java/gov/va/api/health/vistafhirquery/mockservices/LhsLighthouseRpcGatewayMocksV1.java [hunk 2]
```diff
@@ -34,38 +38,85 @@
...
+                matcher -> {
+                  String[] separated = matcher.split("\\^", 4);
+                  if (separated.length == 4) {
+                    var file = separated[0];
+                    var index = separated[1];
+                    if (isNumeric(file) && isNumeric(index)) {
+                      return asFilemanEntry(file, index).toString();
+                    }
+                  }
+                  return null;
```
Comment (jhulbert-va) [inline vista-fhir-query-mock-services/src/main/java/gov/va/api/health/vistafhirquery/mockservices/LhsLighthouseRpcGatewayMocksV1.java:109]: I think we want a fileman value or we throw. The expectation here is to support only the resquests we expect to see.

File: vista-fhir-query-mock-services/src/main/resources/lhslighthouserpcgateway/response-insurance-buffer-coverage-create.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+  "results" : [
```
Comment (jhulbert-va) [inline vista-fhir-query-mock-services/src/main/resources/lhslighthouserpcgateway/response-insurance-buffer-coverage-create.json:2]: I think you can drop this file now.

File: vista-fhir-query-mock-services/src/main/java/gov/va/api/health/vistafhirquery/mockservices/LhsLighthouseRpcGatewayMocksV1.java [hunk 4]
```diff
@@ -34,38 +37,79 @@
...
+                entry -> {
+                  String[] separated = entry.split("\\^", 4);
+                  var file = separated[0];
+                  var index = separated[1];
+                  return asFilemanEntry(file, index);
+                })
+            .distinct()
+            .collect(Collectors.toList());
+    if (results.isEmpty()) {
+      throw new IllegalStateException("No response for: " + ap);
```
Comment (jhulbert-va) [inline vista-fhir-query-mock-services/src/main/java/gov/va/api/health/vistafhirquery/mockservices/LhsLighthouseRpcGatewayMocksV1.java:108]: `Failed to build response for` gives a better sense I think.

File: vista-fhir-query-mock-services/src/main/java/gov/va/api/health/vistafhirquery/mockservices/LhsLighthouseRpcGatewayMocksV1.java [hunk 5]
```diff
@@ -34,38 +37,79 @@
...
+    }
+    return json(results);
```
Comment (jhulbert-va) [inline vista-fhir-query-mock-services/src/main/java/gov/va/api/health/vistafhirquery/mockservices/LhsLighthouseRpcGatewayMocksV1.java:110]:
```
var response = LhsLighthouseRpcGatewayResponse.Results.builder().results(results).build();
return json(response);
```

File: vista-fhir-query-tests/src/test/java/gov/va/api/health/vistafhirquery/tests/r4/CoverageIT.java [hunk 6]
```diff
@@ -30,7 +30,8 @@
   @Test
   void createAndRead() {
     // Requires LHS LIGHTHOUSE RPC GATEWAY to be deployed to vista
-    assumeEnvironmentIn(Environment.QA, Environment.STAGING_LAB, Environment.LAB);
+    assumeEnvironmentIn(
```
Comment (jhulbert-va) [inline vista-fhir-query-tests/src/test/java/gov/va/api/health/vistafhirquery/tests/r4/CoverageIT.java:33]: `assumEnvironmentNotIn(STAGING, PRODUCTION)`

**Review by eclendenning-va — APPROVED** [review]

**Pull request merged**